web/base_viewer.js

```diff
@@ -292,6 +298,25 @@ class BaseViewer {
     if (resetCurrentPageView) {
       this._resetCurrentPageView();
     }
+
+    const pageView = this._pages[val - 1];
+    const pageOpenEvent = {
+      source: this,
+      pageNumber: val,
+    };
+    if (pageView.renderingState === RenderingStates.FINISHED) {
+      this.eventBus.dispatch("pageopen", pageOpenEvent);
+    } else {
+      const onAfterDraw = evt => {
+        if (evt.pageNumber !== val) {
+          return;
+        }
+        this.eventBus._off("pagerendered", onAfterDraw);
+        this.eventBus.dispatch("pageopen", pageOpenEvent);
+      };
+      this.eventBus._on("pagerendered", onAfterDraw);
```

Comment:
This is generally problematic, since there's absolutely no guarantee that the "new" page will ever be rendered; e.g. when scrolling *quickly* through a long document.

Hence you can easily end up in situation where you're have a *very large* number of "pagerendered" event listeners that won't ever get removed.

Reply:
Any hints or ideas to avoid such situation ?